governator-core/src/main/java/com/netflix/governator/conditional/Conditional.java

```diff
@@ -0,0 +1,37 @@
+package com.netflix.governator.conditional;
+
+/**
+ * Contract for any conditional that may be applied to a conditional binding
+ * bound via {@link ConditionalBinder}.
+ * 
+ * Dependencies needed by a concrete conditional should be injected using 
+ * member injection.
+ * 
+ */
+public abstract class Conditional {
```

Comment:
```
public enum Conditional {;
  public static Conditional not(Conditional conditional) {
    return new NotConditional(conditional);
  }
}

interface ConditionalMatcher {
  boolean matches(Injector injector);
  ConditionalMatcher and(ConditionalMatcher conditional);
  ConditionalMatcher or(ConditionalMatcher conditional);
}
```

Is what I had in mind.  It seems to me that and/or could also be static methods and pulled off onto the enum and turn the ConditionalMatcher into a single method interface for lambda goodness.  Your builder would then be like.
```
Conditional.not(Conditional.and(new ConditionA(), Conditional.or(new ConditionB(), new ConditionC())));
```